File: Web Application/backend/models/Item.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

// define a new schema for item
const itemSchema = new Schema({
  itemName: {
    type: String,
    required: true,
  },
  supplierId: {
    type: String,
    required: true,
  },
  unitPrice: {
    type: Number,
    required: true,
  },
  uom: {
    type: String,
    required: false,
  },
});

// create a model based on the item schema
const Item = mongoose.model("Item", itemSchema);

// export the item model to be used in other parts of the application
module.exports = Item;

File: Mobile Application/procure_mate/lib/models/item.dart
import 'package:mongo_dart/mongo_dart.dart';

class Item {
  final ObjectId id;
  final String itemCode;
  final String itemName;
  final String supplierId;
  final double unitPrice;
  final String uom;

  Item(this.id, this.itemCode, this.itemName, this.supplierId, this.unitPrice,
      this.uom);
}
